clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java

```diff
@@ -65,16 +67,59 @@ protected void maybeThrowAuthFailure(Node node) {
         networkClientDelegate.maybeThrowAuthFailure(node);
     }
 
+    /**
+     * Request that a fetch request be issued to the cluster to pull down the next batch of records.
+     *
+     * <p/>
+     *
+     * The returned {@link CompletableFuture} is {@link CompletableFuture#complete(Object) completed} when the
+     * fetch request(s) have been created and enqueued into the network client's outgoing send buffer.
+     * It is <em>not completed</em> when the network client has received the data.
+     *
+     * @return Future for which the caller can wait to ensure that the requests have been enqueued
+     */
+    public CompletableFuture<Void> requestFetch() {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+
+        if (pendingFetchRequestFuture != null) {
+            // In this case, we have an outstanding fetch request, so chain the newly created future to be
+            // invoked when the outstanding fetch request is completed.
+            pendingFetchRequestFuture.whenComplete((value, exception) -> {
+                if (exception != null) {
```

Comment:
Given that you only complete the pending event successfully in the finally block of `poll()`, you could probably just `future.complete(null)`.

Reply:
Just to make sure I understand: are you stating that because the _pendingFetchRequestFuture_ is only ever `complete()`-ed and not `completeExceptionally()`-ed, that the code in `requestFetch()` could be simplified to:

```suggestion
if (pendingFetchRequestFuture != null) {
    // In this case, we have an outstanding fetch request, so chain the newly created future to be
    // completed when the outstanding fetch request is completed.
    pendingFetchRequestFuture.whenComplete((__, ___) -> future.complete(null));
} else {
    pendingFetchRequestFuture = future;
}
```

Reply:
Yes.

Reply:
FYI: I updated the code to catch spurious errors from `prepareFetchRequests()` that could result in needing to complete the chained `CompletableFuture` with an error.